packages/jest-cli/src/jest.js

```diff
@@ -290,175 +48,40 @@ const runCLI = (
 
   readConfig(argv, root)
     .then(config => {
-      if (argv.debug) {
-        /* $FlowFixMe */
-        const testFramework = require(config.testRunner);
-        pipe.write('jest version = ' + VERSION + '\n');
-        pipe.write('test framework = ' + testFramework.name + '\n');
-        pipe.write('config = ' + JSON.stringify(config, null, '  ') + '\n');
-      }
-      if (argv.watch || argv.watchAll) {
-        setWatchMode(argv, argv.watch ? 'watch' : 'watchAll', {
-          pattern: argv._,
-        });
-
-        let currentPattern = '';
-        let hasSnapshotFailure = false;
-        let isEnteringPattern = false;
-        let isRunning = false;
-        let testWatcher;
-        let timer;
-        let displayHelp = true;
-
-        const writeCurrentPattern = () => {
-          clearLine(pipe);
-          pipe.write(chalk.dim(' pattern \u203A ') + currentPattern);
-        };
-
-        const startRun = (overrideConfig: Object = {}) => {
-          if (isRunning) {
-            return null;
-          }
+      createDirectory(config.cacheDirectory);
+      const jestHasteMap = Runtime.createHasteMap(config, {
```

Comment:
`packages/jest-cli/src/__tests__/SearchSource-test.js` is using it, so I didn't remove it. If you prefer, I can also just move it inside of `SearchSource-test.js` as a helper function only for those tests.